src/corehost/cli/deps_resolver.cpp

```diff
@@ -521,31 +521,27 @@ bool deps_resolver_t::resolve_tpa_list(
         get_dir_assemblies(m_app_dir, _X("local"), &items);
     }
 
-    // Probe FX deps entries after app assemblies are added.
-    for (int i = 1; i < m_fx_definitions.size(); ++i)
+    // If additional deps files were specified that need to be treated as part of the
+    // application, then add them to the mix as well.
+    for (const auto& additional_deps : m_additional_deps)
     {
-        // A minor\major roll-forward affects which layer wins
-        bool is_minor_or_major_roll_forward = m_fx_definitions[i]->did_minor_or_major_roll_forward_occur();
-
-        const auto& deps_entries = m_is_framework_dependent ? m_fx_definitions[i]->get_deps().get_entries(deps_entry_t::asset_types::runtime) : empty;
-        for (const auto& entry : deps_entries)
+        auto additional_deps_entries = additional_deps->get_entries(deps_entry_t::asset_types::runtime);
+        for (auto entry : additional_deps_entries)
         {
-            if (!process_entry(m_fx_definitions[i]->get_dir(), entry, i, is_minor_or_major_roll_forward))
+            if (!process_entry(m_app_dir, entry, 0))
             {
                 return false;
             }
         }
     }
 
-    // If additional deps files were specified that need to be treated as part of the
-    // application, then add them to the mix as well.
-    for (const auto& additional_deps : m_additional_deps)
+    // Probe FX deps entries after app assemblies are added.
```

Comment:
Note I reverted the order of probing additional-deps to how the code was before 2.1. This was changed in 2.1 to support issues with additional-deps and roll-forward. However, since we always compare versions now the change made in 2.1 is no longer necessary.